### PR TITLE
Add basic Sessions, Web Share Target Level 2 and other features.

### PR DESCRIPTION
**Pull request by Wojbie**
New Features:

- Audio players added to gallery. (Audio files now automatically show in gallery page)
- If file is not supported by gallery (and shown using filter) as fallback first 256 bytes of file are shown as preview.
- Copy to clipboard buttons on success/error pages.
- Share buttons (in supporting browsers)
- Web Share Target Level 2 - Ability to send files to shitty using phone native share dialog. ( Chrome Canary only amt)
- Notifications support (For now only used by Web Share Target)
- Manifest and Service worker added for use with above features and ready for future expansion ( expanded caching? )

Fixes:

- Brings back Sessions with basic file store and 2 months (untouched) session storage time.
- Copy to clipboard buttons now all work.
- Some handlebars renders were not getting data they needed causing errors in pages.
- Properly wipes corrupted nonces.json and stats.json and reports that to console.

Sidenote:
Version bumped in package.json due to change in stats.json format that will require recreating cache after update.

Also includes:
3 variants of poop file and egg.

Images:
SvVC
0mnj
KFoN
Phone:
tYA9

### CODE REVIEW

File: views/partials/filters.hbs [hunk 1]
```diff
@@ -29,6 +29,10 @@
 			<div class="ui field input">
 				<input type="text" placeholder="search" id="filter-search">
 			</div>
+			<br/>
+			<div onclick="$('#filter-search').val('.jpeg,.jpg,.png,.gif')" class="ui mini icon button"><i class="file image outline icon"></i> Image</div>
+			<div onclick="$('#filter-search').val('.mp3,.wav,.flac')" class="ui mini icon button"><i class="file audio outline icon"></i> Audio</div>
+			<div onclick="$('#filter-search').val('.mp4,.webm')" class="ui mini icon button"><i class="file video outline icon"></i> Video</div>
```
Comment (Lemmmy) [inline views/partials/filters.hbs]: Perhaps wrap these buttons in one element and combine them with the textbox? Or at least something to make them look less cramped up against the textbox

Here's an idea (making the input field 'action mini' and putting buttons inside it)



```html
			<div class="ui field action input mini">
				<input type="text" placeholder="search" id="filter-search">
				<div onclick="$('#filter-search').val('.jpeg,.jpg,.png,.gif')" class="ui mini icon button"><i class="file image outline icon"></i> Image</div>
				<div onclick="$('#filter-search').val('.mp3,.wav,.flac')" class="ui mini icon button"><i class="file audio outline icon"></i> Audio</div>
				<div onclick="$('#filter-search').val('.mp4,.webm')" class="ui mini icon button"><i class="file video outline icon"></i> Video</div>
			</div>
```

File: views/gallery.hbs [hunk 2]
```diff
@@ -49,4 +56,22 @@
 			</div>
 		</div>
 	</div>
-</div>
\ No newline at end of file
+</div>
+{{#section "script"}}
+<script>
+	function fallbackLoad(t){
+		$.ajax({url: t.attr("data-original"), headers: {"Range": "bytes=0-255"}, success: function(result){
```
Comment (Lemmmy) [inline views/gallery.hbs]: This has the potential to error on empty files with a 416 (Requested Range Not Satisfiable) error

Comment (Wojbie) [reply views/gallery.hbs]: Yes it does return 416 on empty files. Should I add some kind of special handling for those? Currently it just don't show anything.

Comment (Lemmmy) [reply views/gallery.hbs]: Maybe some "Empty file" or "preview not shown" text?

**Comment by Wojbie** [comment]
That should resolve all the issued.
Only thing to note is that express.static will always error ito the log when it hits 416 (Requested Range Not Satisfiable) error. This is nothing to worry about (it don't cause anything and end server still receives correct 416 information but it is something to note that will happen in edge case (looking at empty non media file in gallery).

**Comment by Lemmmy** [comment]
Ready to merge when you are

**Pull request merged**